### PR TITLE
[v0.31] Feat/price source type proper [final]

### PR DESCRIPTION
**Pull request by petioptrv**
**Before submitting this PR, please make sure**:
- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue: #1919
- Related Clubhouse Story:

**A description of the changes proposed in the pull request**:

### CODE REVIEW

**Comment by petioptrv** [comment]
@Nullably, here it is. This is ready to merge

File: test/test_pmm.py [hunk 1]
```diff
@@ -620,6 +679,18 @@ def test_inventory_skew_multiple_orders_status(self):
...
+        strategy.asset_price_delegate = self.order_book_asset_del
+        self.clock.add_iterator(strategy)
+        self.clock.backtest_til(self.start_timestamp + 1)
+
+        bid = self.order_book_asset_del.get_price_by_type(PriceType.BestBid)
+        ask = self.order_book_asset_del.get_price_by_type(PriceType.BestAsk)
+        mid_price = self.order_book_asset_del.get_price_by_type(PriceType.MidPrice)
+
+        self.assertEqual((bid + ask) / 2, mid_price)
+
```
Comment (Nullably) [inline test/test_pmm.py:693]: @petioptrv Can you please add a test on `order_book_asset_del.get_price_by_type(PriceType.LastTrade)` here? 
You can simulate trades on the underlying order book, similar to how you did it for `test_basic_one_level_price_type`.
Thanks

**Review by Nullably — CHANGES_REQUESTED** [review]

File: test/test_pmm.py (hunk 1 above)
Comment (petioptrv) [reply test/test_pmm.py:703]: @Nullably I see, test case added

Comment (petioptrv) [reply test/test_pmm.py:703]: @Nullably, test case added

**Review by Nullably — APPROVED** [review]
LGTM

**Pull request merged**